test/brsTypes/components/RoArray.test.js

```diff
@@ -663,6 +728,54 @@ describe("RoArray", () => {
                 expect(src.elements).toEqual([i2, i1, a, b, a1, a2, a3, ar]);
             });
 
+            it("sorts the array based on AA valid mixed field", () => {
+                let a3 = new RoAssociativeArray([
+                    { name: new BrsString("id"), value: new Int32(3) },
+                    { name: new BrsString("name"), value: new BrsString("Betty") },
+                    { name: new BrsString("data"), value: new BrsString("abc") },
+                ]);
```

Comment:
I was a little surprised to see `a3` declared before `a1` and `a2` — is there any particular reason for that?